### DISCUSSION TITLE
the conf "REL_DIR"is not work in version3.9.11

### DISCUSSION THREAD

**Discussion by jacylb** [rabbitmq/rabbitmq-server#4433] [Other] [answered]
I used the “ERL_DIR=/usr/erlang/bin/” in file rabbitmq-defaults of version3.7.14，now I use the 3.9.11，and echo “ERL_DIR=/usr/erlang24.2/bin/” to rabbitmq-defaults， but getting starting error : erl: command not found。I don‘t want to change the $PATH，How could I do

**Comment by michaelklishin** [comment, marked as answer]
You are supposed to modify `rabbitmq-defaults`. 

Modify the `PATH` varaible, `ERL_DIR` was never recommended by our team.

**Reply by jacylb** [reply]
sorry，I wrote the wrong file name，but I am supposed to modify，it doesn‘t work .As shown in figure
![image](https://user-images.githubusercontent.com/102955390/161552972-b87e8801-7942-4cb1-a9ff-0e2cf7673893.png)

**Reply by michaelklishin** [reply]
I'm not sure what you are trying to convey with this screenshot. Like I said, `ERL_DIR` or direct RabbitMQ script modifications is not how you are supposed to configure desired Erlang path, the `PATH` variable is.

See https://github.com/rabbitmq/rabbitmq-server/pull/2461 for an explanation why `ERL_DIR` is no longer used in modern RabbitMQ releases. In fact, it started with `3.8.11` (https://github.com/rabbitmq/rabbitmq-server/commit/68c30553ccf306325a64b1fe6069e6bcc9c26b41) so you must have a leftover in your installation one way or another.

[`rabbitmq-env.conf`](https://rabbitmq.com/configure.html#customise-environment) is loaded early enough that maybe it can be used to export `PATH` but it should not be particularly difficult to do any other way. Just like using `kerl` should be.

**Reply by jacylb** [reply]
thank you.I just found the file `.rabbitmq-server` of 3.7.14 is "exec ${ERL_DIR}erl \"，but the 3.9.11 is “exec erl \”.I also need to modify `.rabbitmq-server`.

**Reply by michaelklishin** [reply]
Then consider yourself running a fork that may or may not be upgradeable. You are doing it wrong, modifying `PATH` is a drastically safer and more predictable option.
